src/blocks/events/event-item/edit.js

```diff
@@ -0,0 +1,137 @@
+/**
+ * External dependencies.
+ */
+import classnames from 'classnames';
+
+/**
+ * Internal dependencies.
+ */
+import { hasEmptyAttributes } from '../../../utils/block-helpers';
+import applyWithColors from './colors';
+import InspectorControls from './inspector';
+
+/**
+ * WordPress dependencies.
+ */
+import { __ } from '@wordpress/i18n';
+import { RichText } from '@wordpress/block-editor';
+import { Component, Fragment } from '@wordpress/element';
+import { compose } from '@wordpress/compose';
+
+const isEmpty = attributes => {
+	const attributesToCheck = [ 'title', 'description', 'eventDay', 'eventMonth', 'eventYear', 'eventTime', 'eventLocation' ];
+	const newAttributes = Object.entries( attributes ).filter( ( [ key ] ) =>
+		attributesToCheck.includes( key )
+	);
+
+	return hasEmptyAttributes( Object.fromEntries( newAttributes ) );
+};
+
+class EventsEdit extends Component {
+	componentDidUpdate() {
+		if ( ( this.props.attributes.externalChange ) ) {
+			this.props.setAttributes( { textColor: this.props.attributes.textColor, externalChange: false } );
+			this.props.setTextColor( this.props.attributes.textColor );
+		}
+	}
+
+	render() {
+		const { className, attributes, setAttributes, textColor } = this.props;
+
+		const textStyles = {
+			color: textColor.color,
+		};
+
+		return (
+			<Fragment>
+				<InspectorControls { ...this.props }
+				/>
+				<div
+					data-page={ String( attributes.pageNum ) }
```

Comment:
This is unnecessary. We can calculate pages and perform pagination dynamically without hard coding these values.